pages/blog/index.tsx

```diff
@@ -59,6 +62,34 @@ export default function BlogIndexPage() {
   const description = 'Find the latest and greatest stories from our community';
   const image = '/img/social/blog.webp';
 
+  const handlePageChange = (pageNumber: number) => {
+    setCurrentPage(pageNumber);
+    if (pageNumber > pageRange[1] - 1) {
+      setPageRange([pageRange[0] + 3, pageRange[1] + 3]);
+    } else if (pageNumber < pageRange[0] + 1) {
+      const newStart = Math.max(pageRange[0] - 3, 1);
+      const newEnd = newStart + 9;
+
+      setPageRange([newStart, newEnd]);
+    }
```

Comment:
Can you provide an explanation of the sequence of events once this function (`handlePageChange`) gets triggered?